Gems/Multiplayer/Code/Source/Pipeline/NetworkPrefabProcessor.cpp

```diff
@@ -102,34 +102,37 @@ namespace Multiplayer
 
         sourceInstance->GetEntities([networkInstance, rootSourceInstance, rootNetworkInstance, prefabName, uniqueName, &context, networkSpawnableAssetId](AZStd::unique_ptr<AZ::Entity>& sourceEntity)
         { 
-            if (sourceEntity->FindComponent<NetBindComponent>())
+            if (!sourceEntity->FindComponent<NetBindComponent>())
             {
-                AZ::Entity* entity = sourceEntity.get();
-                AZ::Entity* netEntity = SpawnableUtils::CreateEntityAlias(
-                    prefabName,
-                    *rootSourceInstance,
-                    uniqueName,
-                    *rootNetworkInstance,
-                    *networkInstance,
-                    entity->GetId(),
-                    AzToolsFramework::Prefab::PrefabConversionUtils::EntityAliasType::Replace,
-                    AzToolsFramework::Prefab::PrefabConversionUtils::EntityAliasSpawnableLoadBehavior::DependentLoad,
-                    NetworkEntityManager::NetworkEntityTag,
-                    context);
-
-                AZ_Assert(
-                    netEntity, "Unable to create alias for entity %s [%zu] from the source prefab instance", entity->GetName().c_str(),
-                    aznumeric_cast<AZ::u64>(entity->GetId()));
-
-                netEntity->InvalidateDependencies();
-                netEntity->EvaluateDependencies();
-
-                PrefabEntityId prefabEntityId;
-                prefabEntityId.m_prefabName = sourceEntity->GetName();
-                NetBindComponent* netBindComponent = netEntity->FindComponent<NetBindComponent>();
-                netBindComponent->SetPrefabAssetId(networkSpawnableAssetId);
-                netBindComponent->SetPrefabEntityId(prefabEntityId);
+                return true; // return true tells GetEntities to continue iterating over all the entities
             }
+
+            AZ::Entity* entity = sourceEntity.get();
+            AZ::Entity* netEntity = SpawnableUtils::CreateEntityAlias(
+                prefabName,
+                *rootSourceInstance,
+                uniqueName,
+                *rootNetworkInstance,
+                *networkInstance,
+                entity->GetId(),
+                AzToolsFramework::Prefab::PrefabConversionUtils::EntityAliasType::Replace,
+                AzToolsFramework::Prefab::PrefabConversionUtils::EntityAliasSpawnableLoadBehavior::DependentLoad,
+                NetworkEntityManager::NetworkEntityTag,
+                context);
+
+            AZ_Assert(
+                netEntity, "Unable to create alias for entity %s [%zu] from the source prefab instance", entity->GetName().c_str(),
```

Comment:
do we need to include source prefab instance identifier here for debugging?

Reply:
Good catch! I've added the source prefab name here :)